pkg/ds/farm.go

```diff
@@ -262,8 +262,13 @@ func (dsf *Farm) handleDSChanges(changes dsstore.WatchedDaemonSets, quitCh <-cha
 		return
 	}
 
+	if changes.Total == 0 {
+		dsf.logger.Errorf("No daemon sets present, asked to delete %d; farm refuses to delete its %d daemon sets", len(changes.Deleted), len(dsf.children))
```

Comment:
actually, this troubles me. This causes the farm's state to become out of sync with the watch's state. By this I mean: The watch will assume that the daemon set farm has processed the deletion of all its daemon sets, but the daemon set farm has, in fact, not.

This means that if the same watch perceives an update to an existing DS, the watch will say it's a new DS (assuming the farm has deleted it). this may have undesirable effects. But does that mean we should terminate the watch instead? Or somehow inform the watch that the changes were rejected, so it keeps its old state?